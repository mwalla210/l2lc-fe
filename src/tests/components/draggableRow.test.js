import React from 'react'
import renderer from 'react-test-renderer'
import DraggableRow from '../../components/draggableRow'

const defaultOptions = {
  connectDragSource: input => {return input},
  connectDropTarget: input => {return input},
  index: 0,
  isDragging: false,
  id: 1,
  moverow: jest.fn(),
}
describe('DraggableRow', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <DraggableRow {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders while dragging, with cursor', () => {
    let options = Object.assign({}, defaultOptions)
    options.isDragging = true
    options.style = {}
    const component = renderer.create(
      <DraggableRow {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
