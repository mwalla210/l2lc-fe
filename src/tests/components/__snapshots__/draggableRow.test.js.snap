// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableRow Renders while dragging 1`] = `
<div
  className="rt-tr"
  id={1}
  role="row"
  style={
    Object {
      "cursor": "move",
      "opacity": 0,
    }
  }
/>
`;

exports[`DraggableRow Renders with snapshot 1`] = `
<div
  className="rt-tr"
  id={1}
  role="row"
  style={
    Object {
      "cursor": "move",
      "opacity": 1,
    }
  }
/>
`;
